src/Microsoft.AspNetCore.SignalR.Core/JsonProtocolDependencyInjectionExtensions.cs

```diff
@@ -27,9 +29,10 @@ public static class JsonProtocolDependencyInjectionExtensions
         /// <param name="builder">The <see cref="ISignalRBuilder"/> representing the SignalR server to add JSON protocol support to.</param>
         /// <param name="configure">A delegate that can be used to configure the <see cref="JsonHubProtocolOptions"/></param>
         /// <returns>The value of <paramref name="builder"/></returns>
-        public static ISignalRBuilder AddJsonProtocol(this ISignalRBuilder builder, Action<JsonHubProtocolOptions> configure)
+        public static TBuilder AddJsonProtocol<TBuilder>(this TBuilder builder, Action<JsonHubProtocolOptions> configure) where TBuilder : ISignalRBuilder
         {
-            builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IHubProtocol, JsonHubProtocol>());
+            builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IHubProtocol, JsonHubProtocol>(
```

Comment:
Why the change?

Reply:
Because otherwise reflection and expression trees will be used to call the constructor, which we wanted to avoid on AOT.

Reply:
Nahhh, lets keep it here. That happens once